ra2ce/graph/graph_files/graph_files_collection.py

```diff
@@ -0,0 +1,185 @@
+
+        Returns:
+            bool: True if any graph already read
+        """
+        return any(
+            iter([x for x in self._graph_collection if x.graph is not None]),
+        )
+
+    def get_graph_file(self, graph_file_type: str) -> GraphFileProtocol:
+        """
+        Get the graph (object)
+
+        Args:
+            graph_file_type (str): Type of graph file
+
+        Raises:
+            ValueError: If the graph_file_type is not one of the known types
+
+        Returns:
+            GraphFileProtocol: Graph object of that specific graph_file_type
+        """
+        _gf = next(
+            (
+                gf
+                for gf in self._graph_collection
+                if Path(gf.name).stem == graph_file_type
+            )
+        )
+        if _gf is None:
+            raise ValueError(f"Unknown graph file type {graph_file_type} provided.")
+        return _gf
+
+    def get_graph(self, graph_file_type: str) -> MultiGraph | GeoDataFrame:
+        """
+        Get the graph
+
+        Args:
+            graph_file_type (str): Type of graph file
+
+        Returns:
+            GraphFileProtocol: Graph of that specific graph_file_type
+        """
+        _graph_file = self.get_graph_file(graph_file_type)
+        return _graph_file.graph
+
+    def get_file(self, graph_file_type: str) -> Path | None:
+        """
+        Get the file path to the graph
+
+        Args:
+            graph_file_type (str): Type of graph file
+
+        Returns:
+            Path: Path to the graph file
+        """
+        _graph_file = self.get_graph_file(graph_file_type)
+        return _graph_file.file
+
+    @classmethod
+    def set_files(cls, parent_dir: Path) -> GraphFilesCollection:
+        """
+        Create a new collection with 1 or more graph files that match the default names
+
+        Args:
+            parent_dir (Path): Path of the parent folder in which the files are searched
+
+        Returns:
+            GraphFilesCollection: Collection of graphs
+        """
+        _collection = cls()
+
+        _collection.base_graph.read_graph(parent_dir)
+        _collection.base_graph_hazard.read_graph(parent_dir)
+        _collection.origins_destinations_graph.read_graph(parent_dir)
+        _collection.origins_destinations_graph_hazard.read_graph(parent_dir)
+        _collection.base_network.read_graph(parent_dir)
+        _collection.base_network_hazard.read_graph(parent_dir)
+
+        return _collection
+
+    def set_file(self, file: Path) -> None:
+        """
+        Set a path to a graph via the collection
+
+        Args:
+            file (Path): Path of the graph
+
+        Raises:
+            ValueError: If the graph_file_type is not one of the known types
+        """
+        _gf = next(
+            (gf for gf in self._graph_collection if Path(gf.name).stem == file.stem)
+        )
+        if _gf is None:
+            raise ValueError(f"Unknown graph file {file} provided.")
```

Comment:
You don't need to cast into path, as you know you are representing a filename (stem + extension). Also, you are missing the default value. A test which checks whether the error is being risen will easily proof this.

```suggestion
        _gf = next(
            (gf for gf in self._graph_collection if gf.name == file.name), None
        )
        if _gf is None:
            raise ValueError(f"Unknown graph file {file} provided.")
```

Reply:
Check